code-server: add v4.96.4

A-N-Other opened the pull request:
This hasn't been updated in a while, so bumping to `@4.96.4`. Installs and functioned as expected for me on our `linux-rocky8-icelake` cluster.

Commit:
code-server: bumping to @4.96.4

bernhardkaindl reviewed (dismissed):
@A-N-Other, as this is on based on VS Code, would you be fine with removing the old versions?

Normally, in spack, old versions that are not needed can be marked as deprecated so we can automatically remove them on regular intervals, but as this is an IDE where the old versions are not needed as a dependency, basically the lastest versions should only be needed.

A-N-Other committed:
Removing older versions

A-N-Other commented:
@bernhardkaindl Perfectly fine by me, yeah. Shouldn't need 20+ old versions hanging around. Modified that now and removed the `url_for_version()` call as that's no longer needed.

bernhardkaindl reviewed (approved):
Great, agreed!
